tests/unit/test_python_evaluation.py

```diff
@@ -38,75 +41,89 @@ def target_metrics(scope="module"):
         "precision": pytest.approx(0.26666, TOL),
         "map": pytest.approx(0.23613, TOL),
         "recall": pytest.approx(0.37777, TOL),
+        "auc": pytest.approx(0.75, TOL),
+        "logloss": pytest.approx(0.7835, TOL)
     }
 
 
 @pytest.fixture(scope="module")
 def python_data():
-    rating_true = pd.DataFrame(
-        {
-            DEFAULT_USER_COL: [1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3],
-            DEFAULT_ITEM_COL: [1, 2, 3, 1, 4, 5, 6, 7, 2, 5, 6, 8, 9, 10, 11, 12, 13, 14],
-            DEFAULT_RATING_COL: [5, 4, 3, 5, 5, 3, 3, 1, 5, 5, 5, 4, 4, 3, 3, 3, 2, 1],
-        }
-    )
-    rating_pred = pd.DataFrame(
-        {
-            DEFAULT_USER_COL: [1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3],
-            DEFAULT_ITEM_COL: [3, 10, 12, 10, 3, 5, 11, 13, 4, 10, 7, 13, 1, 3, 5, 2, 11, 14],
-            PREDICTION_COL: [
-                14,
-                13,
-                12,
-                14,
-                13,
-                12,
-                11,
-                10,
-                14,
-                13,
-                12,
-                11,
-                10,
-                9,
-                8,
-                7,
-                6,
-                5,
-            ],
-        }
-    )
-    rating_nohit = pd.DataFrame(
-        {
-            DEFAULT_USER_COL: [1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3],
-            DEFAULT_ITEM_COL: [100] * rating_pred.shape[0],
-            PREDICTION_COL: [
-                14,
-                13,
-                12,
-                14,
-                13,
-                12,
-                11,
-                10,
-                14,
-                13,
-                12,
-                11,
-                10,
-                9,
-                8,
-                7,
-                6,
-                5,
-            ],
-        }
-    )
-    return rating_true, rating_pred, rating_nohit
+    def _generate_python_data(binary_rating=False):
```

Comment:
don't understand why this is needed, instead of directly returning the ratings

Reply:
Because a function with argument `binary_rating` is returned from the fixture to allow testing different types of metrics, rating metric and binary prediction metric, respectively.

Reply:
yeah, but then we are  using it as fixtures. Let me ping you offline

Reply:
See https://alysivji.github.io/pytest-fixures-with-function-arguments.html

Reply:
oh I see what you are doing, sorry I didn't understand it initially. Yeah, you are right, this is correct. I think it is because I didn't see the argument. 
Would it make sense to set explicitly `binary_rating` in all the instances to make it clearer?

Reply:
@miguelgfierro great point - will do that.